soroban-env-host/src/host.rs

```diff
@@ -1250,6 +1250,28 @@ impl Host {
             .get_authenticated_top_authorizations())
     }
 
+    // Returns the authorizations that have been authenticated for the last
+    // contract invocation.
+    //
+    // Authenticated means that either the authorization was authenticated using
+    // the actual authorization logic for that authorization in enforced mode,
+    // or that it was recorded in recording mode and authorization was assumed
+    // successful.
+    #[cfg(any(test, feature = "testutils"))]
+    pub fn get_authenticated_authorizations(
+        &self,
+    ) -> Result<Vec<(ScAddress, Hash, ScSymbol, ScVec)>, HostError> {
```

Comment:
I'm afraid this might end up too misleading and confusing. Since the call stack is ignored here, there will be a lot of ambiguity in terms of what the actual auth trees were. Also this might create a wrong impression that e.g. in order to do something like authorized A->B call one needs to sign A and B and not a tree of A->B.

Reply:
I agree it is imperfect, but it's a lot simpler to understand than top-level authorization, which I've discovered through writing tests I still don't completely understand. I think other developer will hit this problem too.

Being able to assert did an auth happen, without needing to assert the tree it occurred in, seems reasonable, and only possible if we provide visibility into all auths.

Reply:
This change doesn't prevent us from improving this testing story to include the ability to assert on a tree.

Reply:
Yeah, I guess we can update this once more when the tree support is there...

Reply:
Yup. I think if we see folks using this in a way that's bad, when we update with the tree, we can just replace it as a breaking change to the test api. If folks are using it okay, we can probably have side-by-side if we discover between now and then that there's value.